yarn-project/aztec-nr/compressed-string/src/compressed_string.nr

```diff
@@ -0,0 +1,139 @@
+use dep::aztec::types::type_serialization::TypeSerializationInterface;
+use dep::protocol_types::utils::field::field_from_bytes;
+use dep::std;
+
+// A Fixedsize Compressed String.
+// Essentially a special version of Compressed String for practical use.
+struct FixedsizeCompressedString{
+  value: Field
+}
+
+impl FixedsizeCompressedString{
+  pub fn is_eq(self, other: FixedsizeCompressedString) -> bool {
+    self.value == other.value
+  }
+
+  pub fn from_field(input_field: Field) -> Self {
+    Self {value: input_field}
+  }
+
+  pub fn from_string(input_string: str<31>) -> Self {
+    Self {value: field_from_bytes(input_string.as_bytes(), true)}
+  }
+
+  pub fn to_bytes(self) -> [u8; 31] {
+    let mut result = [0; 31];
+    let bytes = self.value.to_be_bytes(31);
+    for i in 0..31 {
+        result[i] = bytes[i];
+    }
+    result
+  }
+
+  pub fn serialize(self) -> [Field; 1] {
+    [self.value]
+  }
+
+  pub fn deserialize(input: [Field; 1]) -> Self {
+    Self { value: input[0] }
+  }
+}
+
+fn deserialize(fields: [Field; 1]) -> FixedsizeCompressedString {
+    FixedsizeCompressedString { value: fields[0] }
+}
+
+fn serialize(value: FixedsizeCompressedString) -> [Field; 1] {
+    value.serialize()
+}
+// Need to figure this out afterwards.
+global FixedsizeCompressedStringSerializationMethods = TypeSerializationInterface {
+  deserialize,
+  serialize,
+};
+
+struct CompressedString<N, M> {
```

Comment:
Could you document this here? I think it would not be clear to aztec-nr users what is this for. (also would document why do we have two N and M generics)